templates/tutorials/tutorial.html

```diff
@@ -0,0 +1,227 @@
+
+    if (dataLayer) {
+      tutorialFeedbackIcons.forEach(icon => {
+        icon.addEventListener('click', e => {
+          const feedbackValue = e.target.getAttribute('data-feedback-value');
+          dataLayer.push({
+            'event' : 'GAEvent',
+            'eventCategory' : 'feedback',
+            'eventAction' : feedbackValue,
+            'eventLabel' : feedbackValue,
+            'eventValue' : undefined
+          });
+
+          tutorialFeedbackOptions.classList.add('u-hide');
+          tutorialFeedbackResult.classList.remove('u-hide');
+        });
+      });
+    }
+  })();
+</script>
+
+<script>
+(function() {
+  function setActiveLink(navigationItems) {
+    navigationItems.forEach(item => {
+      const link = item.querySelector('.p-tutorial__nav-link');
+      if (link.getAttribute('href') === window.location.hash) {
+        item.classList.add('is-active');
+      } else {
+        item.classList.remove('is-active');
+      }
+    });
+  };
+
+  const navigationItems = document.querySelectorAll('.p-tutorial__nav-item');
+  const toggleButton = document.querySelector('.p-tutorial__nav-toggle');
+
+  setActiveLink(navigationItems);
+
+  window.addEventListener('hashchange', e => {
+    e.preventDefault();
+    setActiveLink(navigationItems);
+  });
+
+  sectionIds = [];
+
+  const tutorialSections = document.querySelectorAll('.p-tutorial__content section');
+  tutorialSections.forEach(section => {
+    sectionIds.push(section.id);
+  });
+
+  // Navigate to first tutorial step on load if no URL hash
+  if (!window.location.hash) {
+    const firstSectionLink = document.querySelector('.p-tutorial__nav-link');
+    window.location.hash = firstSectionLink.getAttribute('href');
+  } else {
+    // Redirect #0, #1 etc. to the correct section
+    match = window.location.hash.match(/^#(\d+)$/);
+
+    if (match) {
+      index = parseInt(match[1]);
+      sectionId = sectionIds[index];
+      window.location.hash = '#' + sectionId;
+      window.location.reload();
+    }
+  }
+})();
+</script>
+
+<script>
+  (function() {
+    const polls = document.querySelectorAll('.poll');
+
+    if (dataLayer) {
+      polls.forEach(poll => {
+        const answers = poll.querySelectorAll('[type="radio"]');
+        const pollId = poll.getAttribute('data-poll-name');
+
+        answers.forEach(answer => {
+          answer.addEventListener('change', e => {
+            const answerLabel = document.querySelector('label[for="' + e.target.id + '"]');
+            const eventLabel = answerLabel.innerText;
+            const eventAction = document.getElementById(pollId).innerText;
+            dataLayer.push({
+              'event' : 'GAEvent',
+              'eventCategory' : 'survey',
+              'eventAction' : eventAction,
+              'eventLabel' : eventLabel,
+              'eventValue' : undefined
+            });
+          });
+        });
+      });
+    }
+  })();
+</script>
+
+<script>
```

Comment:
Any reason these are split into four different IIFEs?

Reply:
just for modularity